### PR TITLE
New version of NGOpt

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -2615,7 +2640,42 @@ def _select_optimizer_cls(self) -> base.OptCls:
...
+        if (
+            self.budget is not None
+            and self.fully_continuous
+            and self.budget < self.dimension ** 2 * 2
+            and self.num_workers == 1
+            and not self.has_noise
+            and self.num_objectives < 2
+        ):
+            return MetaModelOpO  # OnePlusOne seems equivalent so far
+        elif self.fully_continuous and self.budget is not None and self.budget < 600:
```
[the remark this answers is not in the document]
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py:2654]: I kept NGOpt14 identical to previous the present PR, 
though I think it was not a good idea to not have this "fully_continuous" condition.
NGOpt15 differs. This is on purpose.
Hopefully NGOpt16 and NGOpt15 are better than NGOpt14.

[the remark this answers is not in the document]
Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py:2654]: (I keep NGOpt14 for the record, though I think there was a bad idea (due to me) in it )

[the remark this answers is not in the document]
Comment (elenaraponi) [reply nevergrad/optimization/optimizerlib.py:2654]: The thing is that if we keep NGopt14 as it is, MetaModel might be chosen when the problem is not fully_continuous. This is why I would "fix" NGOpt14 by adding if self.fully_continuous and maybe stop NGOpt15 at line 2653.
But if you think this is not dangerous, then I am fine with keeping NGOpt14 for the records.

File: nevergrad/optimization/optimizerlib.py [hunk 2]
```diff
@@ -1511,6 +1509,33 @@ def _internal_tell_candidate(self, candidate: p.Parameter, loss: tp.FloatLoss) -
...
+        multivariate_optimizer: tp.Optional[base.OptCls] = None,
+    ) -> None:
+        super().__init__(_MetaModel, locals())
+        self.multivariate_optimizer = multivariate_optimizer
+
+
+MetaModel = ParametrizedMetaModel().set_name("MetaModel", register=True)
+MetaModelOpO = ParametrizedMetaModel(multivariate_optimizer=OnePlusOne).set_name(
+    "MetaModelOpO", register=True
+)
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1536]: Can we avoid Acronym? It tends to become unreadable

**Review by jrapin — APPROVED** [review]

**Pull request merged**